### PR TITLE
[SYCL][NVPTX][NFC] Refactor NVPTX target configuration

### PR DESCRIPTION
**Pull request by bader**
Avoid using `-sycldevice` for configuring NVPTX target.

### CODE REVIEW

**Comment by bader** [comment]
+@Naghasan.

File: llvm/lib/Target/NVPTX/SYCL/LocalAccessorToSharedMemory.cpp [hunk 1]
```diff
@@ -48,7 +48,9 @@ class LocalAccessorToSharedMemory : public ModulePass {
     // Access `nvvm.annotations` to determine which functions are kernel entry
     // points.
     auto NvvmMetadata = M.getNamedMetadata("nvvm.annotations");
-    assert(NvvmMetadata && "IR compiled to PTX must have nvvm.annotations");
+    if (!NvvmMetadata)
```
Comment (premanandrao) [inline llvm/lib/Target/NVPTX/SYCL/LocalAccessorToSharedMemory.cpp:51]: Is this a related change?

**Review by premanandrao — COMMENTED** [review]
Other changes look okay to me.   Could you add tests for this if applicable?

**Comment by bader** [comment]
> Could you add tests for this if applicable?

This patch doesn't add new functionality - it's a refactoring of existing features, so I think regression tests we already have should be enough.

File: llvm/lib/Target/NVPTX/SYCL/LocalAccessorToSharedMemory.cpp (hunk 1 above)
Comment (bader) [reply llvm/lib/Target/NVPTX/SYCL/LocalAccessorToSharedMemory.cpp:51]: Yes. We skipped this pass for modules w/o `-sycldevice` triple component in llvm/lib/Target/NVPTX/NVPTXTargetMachine.cpp and now this pass is executed regardless of the environment component value. SYCL compiler always added `nvvm.annotations` metadata, but it might be missing. In case when this metadata is not emitted, we just skip the transformation instead of crashing (with enabled assertions).

**Review by premanandrao — APPROVED** [review]

**Comment by bader** [comment]
@Naghasan, does it look okay to you?

**Review by Naghasan — APPROVED** [review]
